### PR TITLE
C# Add request object method unit tests

### PR DESCRIPTION
**Pull request by chrisdunelm**
Updates #1872

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/csharp/csharp_library.baseline [hunk 1]
```diff
@@ -2842,6 +2842,60 @@ namespace Google.Example.Library.V1.Tests
...
+                .Returns(new Mock<Labeler.LabelerClient>().Object);
+            mockGrpcClient.Setup(x => x.CreateOperationsClient())
+                .Returns(new Mock<Operations.OperationsClient>().Object);
+            Shelf expectedResponse = new Shelf
+            {
+                ShelfName = new ShelfName("[SHELF_ID]"),
+                Theme = "theme110327241",
+                InternalTheme = "internalTheme792518087",
+            };
+            // TODO: Add verification of request object
```
Comment (jskeet) [inline src/test/java/com/google/api/codegen/testdata/csharp/csharp_library.baseline]: Any reason not to do this right now? We expect the *exact* request object, right?

File: src/test/java/com/google/api/codegen/testdata/csharp/csharp_library.baseline [hunk 2]
```diff
@@ -2842,6 +2842,60 @@ namespace Google.Example.Library.V1.Tests
...
+            mockGrpcClient.Setup(x => x.CreateShelf(It.IsAny<CreateShelfRequest>(), It.IsAny<CallOptions>()))
+                .Returns(expectedResponse);
+            LibraryServiceClient client = new LibraryServiceClientImpl(mockGrpcClient.Object, null);
+            CreateShelfRequest request = new CreateShelfRequest
+            {
+                Shelf = new Shelf(),
+            };
+            Shelf response = client.CreateShelf(request);
+            Assert.Equal(expectedResponse, response);
```
Comment (jskeet) [inline src/test/java/com/google/api/codegen/testdata/csharp/csharp_library.baseline]: Assert.Same in this case, as we don't expect any interference?

File: src/test/java/com/google/api/codegen/testdata/csharp/csharp_library.baseline (hunk 1 above)
Comment (chrisdunelm) [reply src/test/java/com/google/api/codegen/testdata/csharp/csharp_library.baseline]: I'm now working on a PR to verify the request object in every case including this trivial case; is it OK to not make this change here?

File: src/test/java/com/google/api/codegen/testdata/csharp/csharp_library.baseline (hunk 2 above)
Comment (chrisdunelm) [reply src/test/java/com/google/api/codegen/testdata/csharp/csharp_library.baseline]: Good idea, again, please can I make this change in the next PR?

**Comment by chrisdunelm** [comment]
Thanks :)

**Pull request merged**